src/hssm/hssm.py

```diff
@@ -598,8 +609,97 @@ def sample(
                 # if parent already in posterior
                 del self._inference_obj.posterior["rt,response_mean"]
 
+        # Subset data vars in posterior
+        if hasattr(self, "pymc_model") and self._inference_obj is not None:
+            vars_to_keep = set(
+                [var.name for var in getattr(self, "pymc_model").free_RVs]
+            ).intersection(set(list(self._inference_obj["posterior"].data_vars.keys())))
+        else:
+            raise ValueError("Model has not been initialized yet. Something is wrong!")
+
+        setattr(
+            self._inference_obj,
+            "posterior",
+            self._inference_obj["posterior"][list(vars_to_keep)],
+        )
+
```

Comment:
Since we have a separate `.vi()` method here, this function will only return an `InferenceData` type. Let's change the type annotation for the function signature and docstrings as well